File: Warehouse/src/main/java/warehouse/entities/Product.java
package warehouse.entities;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Data
@ToString
@Entity
@Table(name = "products")
@NoArgsConstructor
public class Product implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name= "name", nullable= false)
    @NonNull
    private String name;

    @Column(name= "price", nullable= false)
    private double price;

    @Column(name= "available_quantity", nullable= false)
    private int availableQuantity;
}

File: ProductQuantityService/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ProductQuantityService.Models
{
    [Table("products")]
    public class Product
    {
        [Column("id")]
        public int Id { get; set; }

        [Column("name")]
        public string? Name { get; set; }

        [Column("price")]
        public double Price { get; set; }

        [Column("available_quantity")]
        public int AvailableQuantity { get; set; }
    }
}
